pkg/inventory/inventory-client.go

```diff
@@ -79,79 +62,157 @@ func NewInventoryClient(factory util.Factory) (*ClusterInventoryClient, error) {
+	if object.SetEquals(objs, clusterObjs) {
+		klog.V(4).Infof("applied objects same as cluster inventory: do nothing")
+		return nil
 	}
-	prevInventoryObjs, err := cic.retrievePreviousInventoryObjects(current, label)
+	clusterInv, err := cic.getClusterInventoryInfo(localInv)
 	if err != nil {
-		return nil, err
+		return err
+	}
+	wrappedInv := WrapInventoryObj(clusterInv)
+	if err = wrappedInv.Store(objs); err != nil {
+		return err
 	}
-	// Remove the current inventory info from the previous inventory infos.
-	pastInventoryInfos := []*resource.Info{}
-	for _, pastInfo := range prevInventoryObjs {
-		past, err := infoToObjMetadata(pastInfo)
+	if !cic.dryRun {
+		clusterInv, err = wrappedInv.GetObject()
 		if err != nil {
-			return nil, err
+			return err
 		}
-		if !current.Equals(past) {
-			pastInventoryInfos = append(pastInventoryInfos, pastInfo)
+		klog.V(4).Infof("replace cluster inventory: %s/%s", clusterInv.Namespace, clusterInv.Name)
+		klog.V(4).Infof("replace cluster inventory %d objects", len(objs))
+		if err := cic.applyInventoryObj(clusterInv); err != nil {
+			return err
 		}
 	}
-	return pastInventoryInfos, nil
+	return nil
+}
+
+// GetClusterObjs returns the objects stored in the cluster inventory object, or
+// an error if one occurred.
+func (cic *ClusterInventoryClient) GetClusterObjs(localInv *resource.Info) ([]object.ObjMetadata, error) {
+	var objs []object.ObjMetadata
+	clusterInv, err := cic.getClusterInventoryInfo(localInv)
+	if err != nil {
+		return objs, err
+	}
+	// First time; no inventory obj yet.
+	if clusterInv == nil {
+		return []object.ObjMetadata{}, nil
+	}
+	wrapped := WrapInventoryObj(clusterInv)
+	return wrapped.Load()
 }
 
-// retrievePreviousInventoryObjects requests the previous inventory objects
-// using the inventory label from the current inventory object. Sets
-// the field "pastInventoryObjects". Returns an error if the inventory
-// label doesn't exist for the current currentInventoryObject does not
-// exist or if the call to retrieve the past inventory objects fails.
-func (cic *ClusterInventoryClient) retrievePreviousInventoryObjects(current *object.ObjMetadata, label string) ([]*resource.Info, error) {
-	if cic.retrievedInventoryObjects {
-		return cic.pastInventoryObjects, nil
+// getClusterInventoryObj returns a pointer to the cluster inventory object, or
+// an error if one occurred. Returns the cached cluster inventory object if it
+// has been previously retrieved. Uses the ResourceBuilder to retrieve the
+// inventory object in the cluster, using the namespace, group resource, and
+// inventory label. Merges multiple inventory objects into one if it retrieves
+// more than one (this should be very rare).
+//
+// TODO(seans3): Remove the special case code to merge multiple cluster inventory
+// objects once we've determined that this case is no longer possible.
+func (cic *ClusterInventoryClient) getClusterInventoryInfo(localInv *resource.Info) (*resource.Info, error) {
```

Comment:
The method name says get but it is deleting past inventory objects silently in the background. Though it is a one time step, we need to think of the failure scenarios. 

Nit: get prefix is not recommended as per go readability. Instead it can just be  ClusterInventoryInfo.